libraries/aws_ec2_instance.rb

```diff
@@ -125,6 +125,10 @@ def role
     end
   end
 
+  def resource_id
+    @instance[:instance_id]
```

Comment:
This is an example of an access that will fail if the lookup has failed. If @instance is nil, then the [] access will throw an exception. Try it like this, with a dummy instance ID like i-12345678:

```
control "ec2 instance-check" do
  describe aws_ec2_instance("i-12345678") do
    it { should exist }
  end
end
```

This throws an exception for me with the json reporter:
```
[cwolfe@lodi temp]$ be inspec exec -t aws:// cloudy --no-create-lockfile --reporter json > aws.json
[2022-05-25T13:56:29-04:00] WARN: AWS Service Error encountered running a control with Resource aws_ec2_instance. Error message: The instance ID 'i-12345678' does not exist. You should address this error to ensure your controls are behaving as expected.
bundler: failed to load command: inspec (/Users/cwolfe/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/bin/inspec)
libraries/aws_ec2_instance.rb:129:in `resource_id': undefined method `[]' for nil:NilClass (NoMethodError)
	from /Users/cwolfe/sandbox/inspec/inspec-5/lib/inspec/reporters/json.rb:59:in `extract_resource_id'
	from /Users/cwolfe/sandbox/inspec/inspec-5/lib/inspec/reporters/json.rb:50:in `block in profile_results'
```